src/plugins/externallights/index.js

```diff
@@ -1,111 +1,176 @@
-          self.settings[1] = setting;
-        } else {
-          // Find the closest level in our map
-          var closest = self.levelMap.reduce(function (prev, curr) {
-              return Math.abs(curr - level) < Math.abs(prev - level) ? curr : prev;
-            });
-          // Set the new setting value based on the index of the closest level
-          self.settings[1] = self.levelMap.indexOf(closest);
+
+        setTargetPower( powerIn )
+        {
+            var self = this;
+
+            // Validate input
+            if( isNaN( powerIn ) )
+            {
+              // Ignore
+              return;
+            }
+
+            // Apply limits
+            if( powerIn > 1.0 )
+            {
+                self.targetPower = 1.0;
+            }
+            else if( powerIn < 0 )
+            {
+                self.targetPower = 0;
+            }
+            else
+            {
+                self.targetPower = powerIn;
+            }
+
+            self.targetPower_enc = encode( self.targetPower );
+
+            // Start targetPower sync, if not already running
+            self.SyncTargetPower.start();
         }
-        deps.cockpit.emit('plugin.externalLights.state', 1, { level: self.settings[1] });
-      }
-    });
-    var adjustLights = function adjustLights(lightNum, value) {
-      // Modify current setting
-      setLights(lightNum, self.settings[lightNum] + value);
-    };
-    var toggleLights = function toggleLights(lightNum) {
-      if (self.settings[lightNum] > 0) {
-        // Set to min power
-        setLights(lightNum, 0);
-      } else {
-        // Set to max power
-        setLights(lightNum, self.maxLevel);
-      }
-    };
-    var setLights = function setLights(lightNum, value) {
-      console.log('Attemping to set lights [' + lightNum + '] to: ' + value);
-      // Range limit the new setting from 0 to the max number of defined levels
-      if (value < 0) {
-        value = 0;
-      } else if (value >= self.maxLevel) {
-        value = self.maxLevel;
-      }
-      // Make sure the new setting is an integer
-      self.settings[lightNum] = Math.round(value);
-      console.log('Setting lights [' + lightNum + '] to: ' + self.settings[lightNum]);
-      var command = 'elight' + lightNum + '(' + self.levelMap[self.settings[lightNum]] + ')';
-      deps.globalEventLoop.emit('mcu.SendCommand', command);
+        
+        start()
+        {
+          this.listeners.settings.enable();
+        }
+
+        stop()
+        {
+          this.listeners.settings.disable();
+          this.listeners.mcuStatus.disable();
+          this.listeners.setTargetPower.disable();
+        }
+
+        getSettingSchema()
+        {
+            //from http://json-schema.org/examples.html
+            return [{
+                'title': 'External Lights',
+                'type': 'object',
+                'id': 'external-lights',
+                'properties': {
+                    'steps': 
```

Comment:
Removing to use pre-programmed steps to avoid known bad values given the limitations of the hardware.